### PR TITLE
Add FetchJWTSVIDs and SubscribeToJWTBundles

### PR DESCRIPTION
**Pull request by loveyana**
Signed-off-by: Yuhan Li <liyuhan.loveyana@bytedance.com>

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

Delegated Identity API

**Description of change**

Add FetchJWTSVIDs and SubscribeToJWTBundles in Delegated Identity API.

**Which issue this PR fixes**

https://github.com/spiffe/spire/issues/2788

### CODE REVIEW

**Review by amartinezfayo — COMMENTED** [review]
Thanks for this contribution @loveyana!
This is looking great. Could you please update the dependency of `spire-api-sdk` to point to the latest commit of `next` branch? That should clean the PR of build errors.

**Comment by amartinezfayo** [comment]
Hi @loveyana, did you have a chance to look at the comments from @MarcosDY?

File: pkg/agent/api/delegatedidentity/v1/service.go [hunk 1]
```diff
@@ -245,6 +250,126 @@ func (s *Service) SubscribeToX509Bundles(req *delegatedidentityv1.SubscribeToX50
...
+	}
+
+	selectors, err := api.SelectorsFromProto(req.Selectors)
+	if err != nil {
+		log.WithError(err).Error("Invalid argument; could not parse provided selectors")
+		return nil, status.Error(codes.InvalidArgument, "could not parse provided selectors")
+	}
+	var spiffeIDs []spiffeid.ID
+
+	log = log.WithField(telemetry.Registered, true)
```
[the remark this answers is not in the document]
Comment (loveyana) [reply pkg/agent/api/delegatedidentity/v1/service.go]: I temporarily removed the Registered ID, and I looked at the description in telemetry. It is more like whether the caller is authenticated

File: pkg/agent/api/delegatedidentity/v1/service.go [hunk 2]
```diff
@@ -245,6 +250,126 @@ func (s *Service) SubscribeToX509Bundles(req *delegatedidentityv1.SubscribeToX50
...
+		loopLog.WithField(telemetry.TTL, ttl.Seconds()).Debug("Fetched JWT SVID")
+	}
+
+	return resp, nil
+}
+
+func (s *Service) SubscribeToJWTBundles(req *delegatedidentityv1.SubscribeToJWTBundlesRequest, stream delegatedidentityv1.DelegatedIdentity_SubscribeToJWTBundlesServer) error {
+	ctx := stream.Context()
+	log := rpccontext.Logger(ctx)
+	cachedSelectors, err := s.isCallerAuthorized(ctx, log, nil)
```
[the remark this answers is not in the document]
Comment (loveyana) [reply pkg/agent/api/delegatedidentity/v1/service.go:317]: I'm not sure if your comment is 
```
var cachedSelectors []*common.Selector
if cachedSelectors, err: = s.is CallerAuthorized (ctx, log, nil); err != nil {return err}
```
I looked at the spire code and did not recommend such an implementation. Are you sure we need this?

File: pkg/agent/api/delegatedidentity/v1/service_test.go [hunk 3]
```diff
@@ -351,6 +357,205 @@ func TestSubscribeToX509Bundles(t *testing.T) {
...
+		identities     []cache.Identity
+		authSpiffeID   []string
+		audience       []string
+		expectCode     codes.Code
+		expectMsg      string
+		attestErr      error
+		managerErr     error
+		expectTokenIDs []spiffeid.ID
+	}{
+		{
```
[the remark this answers is not in the document]
Comment (loveyana) [reply pkg/agent/api/delegatedidentity/v1/service_test.go:378]: good point

File: pkg/agent/api/delegatedidentity/v1/service_test.go [hunk 4]
```diff
@@ -441,6 +646,16 @@ func (m *FakeManager) SubscribeToCacheChanges(selectors cache.Selectors) cache.S
 	return newFakeSubscriber(m, m.updates)
 }
 
+func (m *FakeManager) FetchJWTSVID(ctx context.Context, spiffeID spiffeid.ID, audience []string) (*client.JWTSVID, error) {
+	svid := m.ca.CreateJWTSVID(spiffeID, audience)
+	if m.err != nil {
```
[the remark this answers is not in the document]
Comment (loveyana) [reply pkg/agent/api/delegatedidentity/v1/service_test.go:686]: done

File: pkg/agent/api/delegatedidentity/v1/service.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (loveyana) [reply pkg/agent/api/delegatedidentity/v1/service.go:317]: I'm not sure if your comment is 
```
var cachedSelectors []*common.Selector
if cachedSelectors, err: = s.is CallerAuthorized (ctx, log, nil); err! = nil {return err} 
```
I looked at the spire code and did not recommend such an implementation. Are you sure we need this?

**Comment by loveyana** [comment]
> Hi @loveyana, did you have a chance to look at the comments from @MarcosDY?

Sorry for the late reply, I have been spending a lot of time on internal matters recently.

File: pkg/agent/api/delegatedidentity/v1/service.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarcosDY) [reply pkg/agent/api/delegatedidentity/v1/service.go:317]: sorry for the confusion... it is basically put error verification together with line where error is returned, 
for example:

```
ctx := stream.Context()
log := rpccontext.Logger(ctx)

cachedSelectors, err := s.isCallerAuthorized(ctx, log, nil)
if err != nil {
	return err
}
```
it is generally written this way to simplify reading

[the remark this answers is not in the document]
Comment (loveyana) [reply pkg/agent/api/delegatedidentity/v1/service.go]: 😯 I do pay little attention to this matter, it has been completed now.

**Review by amartinezfayo — APPROVED** [review]
Thank you @loveyana!

**Pull request merged**